torch/nn/functional.py

```diff
@@ -1130,10 +1134,11 @@ def poisson_nll_loss(input, target, log_input=True, full=False, size_average=Tru
     if full:
         mask = target > 1
         loss[mask] += (target * torch.log(target) - target + 0.5 * torch.log(2 * math.pi * target))[mask]
+    if not reduce:
+      return torch.mean(loss, 1)
```

Comment:
This should just be `return loss`.

I know the documentation for reduce says "loss per batch element" but the behavior of reduce is really putting out a loss per element of the input